entrypoint.sh

```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 
 # Run python script, capturing all stdout as we go
+py_err=$( python /app/lock.py 2>&1 1>log.txt )
```

Comment:
For my own understanding:
- run the python script
- redirect the `stderr` to `stdout`
- redirect `stdout` to the `log.txt` file
- assign something to the `py_err` variable

I'm not sure what the something is though? It would make sense that we want to capture `stderr` to the variable, but hasn't it been redirected to `stdout`? When a stream is redirected, does it still exist at it's origin?

Reply:
pretty much yes. My understanding of redirection is imperfect, so I'm not sure if there's a cleaner way, but essentially this:

* redirects `stdout` to `log.txt` 
* redirects `stderr` to `stdout`, which is then captured in the `$py_err` variable (I believe this var capture technique can only work with `stdout`)
* iterate the log.txt file capturing output (originally `stdout` to the line by line var `$output`) as previous
* later, tests to see if there's anything in the `$py_err` var (`[ -z "$py_err" ]`) and uses that to set the bash exit output, as well as the error message (the contents of the var will be the exception message)

Reply:
Discussed offline and I now understand how it works. Perhaps we should name log.txt to stdout.txt to make it clear?